### ISSUE TITLE
Execution speed and low GPU utilisation

### ISSUE THREAD

**Issue by CihanTopal** [milesial/Pytorch-UNet#127]
I performed tests on different mid-level and high-end GPUS and see that the GPU utilization is around 1-2%. And thus, it takes too long to train the network. As an example, I have a dataset containing ~750K images with 512x424 pix resolution and during training it can process 8-9 images per second on a GTX 1080 Ti card. (scale=1 and batch size=5. With this speed, it takes almost 30 hours to complete the execution. 
So, I wonder if this performance figures are too different than yours, which means I doing sth wrong?

**Comment by milesial** [comment]
The scale plays a big role in this. I never trained with scale=1 because of memory issues. But i suspect your batch size is too low to fully take advantage of the GPU parallelization. You could also try caching images if you have enough memory.

**Comment by CihanTopal** [comment]
Thank you for your answer.
I also tried different scale factors (0.5~1) and batch numbers (1~5). Even in an 11 GB GPU, larger batches than 5 don't work due to the lack of memory. But, regardless of the scale factor and batch size, I could not be able to see a GPU utilization larger than 2% in different machines and GPUs. 
My question is, am I doing something wrong?
Can you please tell me what the utilization in your experiments? Or how many images can be trained per second?

**Comment by milesial** [comment]
In my experiments, the utilization went up to 100% but not at all times, you could tell when the dataloader was loading the images from the disk and when it was passing them trough the net and backwards.

The images per second count was not above 10, but my images were higher quality and I had access to lower RAM.

You might also try to tweak the parameters used for the construction of the dataloader.
